src/sqlfluff/rules/L016.py

```diff
@@ -339,6 +356,7 @@ def generate_fixes_to_coerce(
         fixes = []
         for split, indent in split_at:
             if split.segments:
+                assert indent_section
```

Comment:
The function `generate_fixes_to_coerce()` clearly assumes `indent_section` is not `None`, but that's not clear to mypy. Hence the assertion.